docs/specs/ecosystem-service-credit.md

```diff
@@ -0,0 +1,98 @@
+# Messages
+
+## Credit Issuance, Transfer and Consumption
```

Comment:
This first section seems clear enough.
Mainly notes that make sense after our conversations.

If this is a spec for me to implement, fine. If this is a spec to post to explain the design to others, this needs much more English and not just the code structs